### ISSUE TITLE
Linux Kernel Sockets

### ISSUE THREAD

**Issue by reveng007** [reveng007/reveng_rtkit#2]
Adding Linux Kernel Sockets to this LKM rootkit, so that this rootkit acts as an all-rounder. Both, as a LKM rootkit as well as stealthy C2 Server.

**Comment by loneicewolf** [comment]
posting here as well, just to make sure it comes across - what do you specifically want `kernel sockets` to achieve? :) 
https://github.com/reveng007/reveng_rtkit/issues/11#issue-1559846247

**Comment by reveng007** [comment]
Sorry for being vague.
Actually that would be C2 client, so that our C2 server listening on target gets automatically connected to the lkm whenever it comes online.

Actually, i want to eradicate the use of revshell to get a shell in the victim machine.

Does this sound familiar now?

**Comment by loneicewolf** [comment]
Yes I think that makes sense now, (+1 for mentioning the revshell ) thanks! if I have more questions I'll ask.
